umbra-js/src/utils/cns.ts

```diff
@@ -1,57 +1,100 @@
 /**
  * @dev Functions for interacting with the Unstoppable Domains Crypto Name Service (CNS)
  */
-
-import { BigNumber } from '@ethersproject/bignumber';
+import { computePublicKey } from '@ethersproject/signing-key';
 import { default as Resolution } from '@unstoppabledomains/resolution';
-import type { EthersProvider } from '../types';
-import * as constants from '../constants.json';
-import * as cnsRegistryAbi from '../abi/CnsRegistry.json';
+import type { EthersProvider, TransactionResponse } from '../types';
 import * as cnsResolverAbi from '../abi/CnsResolver.json';
 import { createContract } from './utils';
 
-const { CNS_REGISTRY } = constants;
+export const cnsKeyPathSpending = 'crypto.ETH.umbra.spending_public_key';
+export const cnsKeyPathViewing = 'crypto.ETH.umbra.viewing_public_key';
+
+/**
+ * @notice Returns supported CNS domain endings
+ */
+export const supportedCnsDomains = ['.crypto', '.zil'];
 
-export const umbraKeySignature = 'crypto.ETH.signature-v0';
+/**
+ * @notice Returns true if the provided name is an CNS domain, false otherwise
+ * @param domainName Name to check
+ */
+export function isCnsDomain(domainName: string) {
+  for (const suffix of supportedCnsDomains) {
+    if (domainName.endsWith(suffix)) {
+      return true;
+    }
+  }
+  return false;
+}
 
 /**
  * @notice Computes CNS namehash of the input CNS domain, normalized to CNS compatibility
  * @param name CNS domain, e.g. myname.crypto
  * @param resolution Resolution instance of @unstoppabledomains/resolution
  */
 export function namehash(name: string, resolution: Resolution) {
+  if (typeof name !== 'string') {
+    throw new Error('Name must be a string with a supported suffix');
+  }
+  if (!isCnsDomain(name)) {
+    throw new Error(`Name does not end with supported suffix: ${supportedCnsDomains.join(', ')}`);
+  }
   return resolution.namehash(name);
 }
 
 /**
- * @notice For a given CNS domain, recovers and returns the public key from its signature
+ * @notice For a given ENS domain, returns the public keys, or throws if they don't exist
```

Comment:
This says ENS when it should say CNS, right? Maybe copy pasted from the other class?

Reply:
Yes! Fixed in 1723291